src/ros_filter.cpp

```diff
@@ -813,6 +813,18 @@ namespace RobotLocalization
     nhLocal_.param("dynamic_process_noise_covariance", dynamicProcessNoiseCovariance, false);
     filter_.setUseDynamicProcessNoiseCovariance(dynamicProcessNoiseCovariance);
 
+    std::vector<double> initial_state(STATE_SIZE, 0.0);
+    if(nhLocal_.getParam("initial_state", initial_state))
+    {
+      if(initial_state.size() != STATE_SIZE)
+      {
+        ROS_ERROR_STREAM("Initial state must be of size " << STATE_SIZE << ". Provided config was of "
+          "size " << initial_state.size() << ". The initial state will be ignored.");
```

Comment:
The latter. There should be an else on this inner if. That will also save us from calling setState at all if we don't have one specified, which is slightly preferable.